templates/workers/worker-horizontalpodautoscaler.yaml

```diff
@@ -31,4 +32,39 @@ spec:
       resource:
         name: memory
         targetAverageUtilization: {{ .Values.workers.autoscaling.targetMemoryUtilization }}
+{{- else}}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ .Release.Name }}-worker
+  labels:
+    tier: airflow
+    component: worker-horizontalpodautoscaler
+    release: {{ .Release.Name }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    heritage: {{ .Release.Service }}
+{{- with .Values.airflow.labels }}
+{{ toYaml . | indent 4 }}
+{{- end }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: StatefulSet
+    name: {{ .Release.Name }}-worker
+  minReplicas: {{ .Values.workers.autoscaling.minReplicas }}
+  maxReplicas: {{ .Values.workers.autoscaling.maxReplicas }}
+  metrics:
+    - type: Resource
+      resource:
+        name: cpu
+        target:
+          type: Utilization
+          averageUtilization: {{ .Values.workers.autoscaling.targetCPUUtilization }}
+    - type: Resource
+      resource:
+        name: memory
+        target:
+          type: Utilization
+          averageUtilization: {{ .Values.workers.autoscaling.targetMemoryUtilization }}
+{{- end }}
```

Comment:
This is too much duplicated code. We should template the apiVersion in [_helpers.tpl](https://github.com/astronomer/airflow-chart/blob/d80292e2d5/templates/_helpers.yaml) and then put the if/else condition around only the parts that are different in the metrics section. There is [something similar in the houston ingress](https://github.com/astronomer/astronomer/blob/4927e02f62/charts/astronomer/templates/houston/ingress.yaml#L43-L64) that you can use as reference.